src/org/javarosa/core/model/FormIndex.java

```diff
@@ -90,7 +91,7 @@
  * @author Clayton Sims
  *
  */
-public class FormIndex implements Serializable {
+public final class FormIndex implements Serializable {
```

Comment:
Now I realize that this change could break downstream client code cause this class is public. I'm OK to set it `final`, but I just wanted to point it out.

Reply:
Oh yes, I should have mentioned that. It does make `equals` a little more correct in the sense that it doesn't have to deal with subclasses. But really, let's not bother doing that here. I've removed it and will let you merge if you're otherwise satisfied.